pkgs/applications/editors/rstudio/update.sh

```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env nix-shell
+#!nix-shell -i bash -p nix-update yarn2nix
+
+set -euo pipefail
+
+nixpkgsTop="$(git rev-parse --show-toplevel)"
+rstudioDir="$nixpkgsTop/pkgs/applications/editors/rstudio"
+
+nix-update rstudio
```

Comment:
You can grab the latest rstudio version with GitHub's API with

```sh
version=$(curl --silent https://api.github.com/repos/rstudio/rstudio/releases/latest | jq --raw-output '.tag_name')
```

And not use `nix-update`, but write the sha256 of the src to an external file and read it in the `default.nix` expression with:

```
nix shell nixpkgs#nix-prefetch-git -c nix-prefetch-git https://github.com/rstudio/rstudio --rev ${version} > $dirname/sha256
```

What's interesting though, is that the `curl` command above doesn't list any [published releases](https://stackoverflow.com/a/26146274/4935114), and using:

```
curl --silent https://api.github.com/repos/rstudio/rstudio/tags | jq '.[0]'
```

Doesn't return the newest tag with the new version scheme, has't yet figure out why..

Reply:
@Artturin Perhaps this will work for us, at least until they'll make a major change in their website:

```sh
curl --silent 'https://www.rstudio.com/products/rstudio/download/' | pup ':parent-of(h3:contains("Source Code")) > a  > attr{href}' | awk -F/ '{gsub(/^v/, "", $NF); print $NF}'
```